### PR TITLE
Replace usage of setTimeout with step_timeout in websockets

### PR DESCRIPTION
**Pull request by gsnedders**
This is expected to help test stability on slow configurations, as the
timeouts will be multiplied by the timeout multiplier.

Split out from #1816.

### CODE REVIEW

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/204278553)

# Firefox (nightly channel) #
Testing web-platform-tests at revision a9d733d076d8df73aa22ffcc7f1737b907fb15bc
Using browser at version BuildID 20170221110300; SourceStamp d84beb192e57e26846c82d3df3599381f4663792
Starting 10 test iterations

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/204278554)

# Chrome (unstable channel) #
Testing web-platform-tests at revision a9d733d076d8df73aa22ffcc7f1737b907fb15bc
Using browser at version 58.0.3013.3 dev
Starting 10 test iterations

**Review by zcorpan — APPROVED** [review]
LGTM but dunno what happened in Travis

**Comment by annevk** [comment]
I think this broke because it needs to be `t.step_timeout` and such.

**Comment by annevk** [comment]
You know what, I'll try to clean this up myself.

**Comment by ricea** [comment]
I've noticed that many of these timeouts seem to be of the form

```javascript
t.step_timeout(() => t.done(), 50);
```

and really shouldn't be needed anyway. This is a pre-existing issue, not a problem with this change. I will clean them up on tests I actually touch.

**Comment by zcorpan** [comment]
That's the intent, yes, and there was originally a bug in Presto that it sometimes happened. So to verify that it never happened, I changed the tests to have this delay (it might have been longer originally but shortened in order to make tests not take too long to run).

However, it seems reasonable to remove the timeout everywhere but one test, assuming that nobody fires too many events currently.

**Comment by ricea** [comment]
@zcorpan Thanks for the rapid response!

My preference would be to have one explicit test for this. When tests test multiple unrelated things it can make them hard to follow, and result in a single bug breaking dozens of tests, potentially hiding other problems.
